source/Octopus.Tentacle.Tests.Integration/TentacleCommandLineTests.cs

```diff
@@ -0,0 +1,659 @@
+                .Should()
+                .Contain(
+                    "configure",
+                    "help",
+                    "run",
+                    "version",
+                    "show-master-key",
+                    "show-thumbprint");
+        }
+
+        [Test]
+        [TentacleConfigurations(testCurrentVersionOnly: true)]
+        public async Task CommandSpecificHelpAsJsonCanBeParsedByAutomationScripts(TentacleConfigurationTestCase tc)
+        {
+            var (_, stdout, stderr) = await RunCommandWithValidExitCodeAssert(tc, "version --help --format=json");
+
+            stderr.Should().BeNullOrEmpty();
+            var help = JsonConvert.DeserializeAnonymousType(
+                stdout,
+                new
+                {
+                    Name = "",
+                    Description = "",
+                    Aliases = Array.Empty<string>(),
+                    Options = new[]
+                    {
+                        new
+                        {
+                            Name = "",
+                            Description = ""
+                        }
+                    }
+                });
+
+            help.Name.Should().Be("version");
+            help.Options.Select(o => o.Name).Should().Contain("format");
+        }
+
+        [Test]
+        [TentacleConfigurations(testCurrentVersionOnly: true)]
+        public async Task CommandSpecificHelpAsJsonLooksSensibleToHumans(TentacleConfigurationTestCase tc)
+        {
+            var (_, stdout, stderr) = await RunCommandWithValidExitCodeAssert(tc, "version --help --format=json");
+            stderr.Should().BeNullOrEmpty();
+
+            stdout.Should().Be(
+@"{
+  ""Name"": ""version"",
+  ""Description"": ""Show the Tentacle version information"",
+  ""Aliases"": [],
+  ""Options"": [
+    {
+      ""Name"": ""format"",
+      ""Description"": ""The format of the output (text,json). Defaults to text.""
+    }
+  ],
+  ""CommonOptions"": [
+    {
+      ""Name"": ""help"",
+      ""Description"": ""Show detailed help for this command""
+    }
+  ]
+}");
+        }
+
+        [Test]
+        [TentacleConfigurations(testCurrentVersionOnly: true)]
+        public async Task HelpForInstanceSpecificCommandsAlwaysWorks(TentacleConfigurationTestCase tc)
+        {
+            var (_, stdout, stderr) = await RunCommand(tc, "help --format=json");
+
+            stderr.Should().BeNullOrEmpty();
+            var help = JsonConvert.DeserializeAnonymousType(
+                stdout,
+                new
+                {
+                    Commands = new[]
+                    {
+                        new
+                        {
+                            Name = "",
+                            Description = "",
+                            Aliases = new string[0]
+                        }
+                    }
+                });
+
+            var failed = help.Commands.Select(async c =>
```

Comment:
nit: Should we verify commands is non zero in size?

Reply:
Done